hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -1023,15 +1028,21 @@ private long sleepWithBusyWait(long timeoutInMillis, long sleep) {
 
     @Override
     public boolean isMemberStateSafe() {
+        return getMemberState() == SAFE;
+    }
+
+    public InternalPartitionServiceState getMemberState() {
         if (hasOnGoingMigrationLocal()) {
-            return false;
+            return MIGRATION_LOCAL;
         }
+
         if (!node.isMaster()) {
             if (hasOnGoingMigrationMaster(Level.OFF)) {
-                return false;
+                return MIGRATION_ON_MASTER;
             }
         }
-        return isReplicaInSyncState();
+
+        return isReplicaInSyncState() ? SAFE : REPLICA_NOT_SYNC;
```

Comment:
I would choose 2 names that are obvious each others opposites. 

e.g. 
REPLICA_NOT_SYNC and REPLICA_SYNC
SAFE and UNSAFE

This makes the code easier to understand because you know they are connected to each other.